dataflow/src/main/java/org/checkerframework/dataflow/analysis/AnalysisResult.java

```diff
@@ -195,6 +201,52 @@ private static void mergeTreeLookup(
         return merged;
     }
 
+    /**
+     * Returns true if the given tree is dead code (will never be executed at run time).
+     *
+     * <p>This method is conservative. If it returns true, the tree is dead code. If it returns
+     * false, the tree may or may not be dead code.
+     *
+     * @param tree a tree
+     * @return true if the tree is dead code
+     */
+    public boolean isDeadCode(Tree tree) {
+        if (!(tree instanceof ExpressionTree)) {
+            return false;
+        }
+        switch (tree.getKind()) {
+            case MEMBER_SELECT:
+                return false;
+            case ASSIGNMENT:
+                // This may be an expression or a statement.
+                return false;
+            case IDENTIFIER:
+            case ARRAY_ACCESS:
+                // There might be a missing "this." prefix.  (Can I test that?)
```

Comment:
I don't understand what a missing "this" prefix is.  Do you mean a `this` tree created by dataflow for a field access?

Reply:
This is explaining why both these cases, and MEMBER_SELECT, are all necessary.  I'll remove it.